### ISSUE TITLE
Generic User Chart

### ISSUE THREAD

**Issue by liversedge** [GoldenCheetah/GoldenCheetah#3344]
Once the Python QT chart integration is done #3325, refactor out the viz parts and create a generic chart that uses user data formulas (from all plot) to create data series and feed to the qt viz chart aspects that have been refactored.

**Comment by liversedge** [comment]
Generic plot created from a refactor of the python chart in commit 70e31235ed8f3d357c3b6c52c9c26ed70177320a

**Comment by liversedge** [comment]
User chart was created in commit 0f1224bb35800d8504869

**Comment by liversedge** [comment]
Need to rework the mechanism for preparing data to show.

I think a similar approach to the user metrics - a set of functions that can be optionally implemented. iterating over rides/samples/measures to aggregate and return as an x and y series.

**Comment by liversedge** [comment]
Additionally, the generic charting needs to offer support from pan/zoom and scroll, annotations, styling and different axis types (date and time most pressing), data labels and combining scatter, bar and line curves on a single chart.

**Comment by liversedge** [comment]
At some point will also need to consider how interval selection and compare mode will interact.

**Comment by liversedge** [comment]
Closing this as most of the heavy lifting is now done.
Will raise a new issue and place into the backlog for possible enhancements.
